src/configuration.ts

```diff
@@ -510,21 +511,35 @@ export function parseConfigFile(
     }
 
     function parse(config: RawConfigFile, dir?: string): IConfigurationFile {
+        const rulesDirectory: string | string[] = getRulesDirectories(config.rulesDirectory, dir);
+        const jsRules = typeof config.jsRules === "boolean" ?
+                            parseRules(config.rules, config.jsRules, rulesDirectory) :
+                            parseRules(config.jsRules);
+
         return {
             extends: arrayify(config.extends),
-            jsRules: parseRules(config.jsRules),
+            jsRules,
             linterOptions: parseLinterOptions(config.linterOptions, dir),
             rules: parseRules(config.rules),
-            rulesDirectory: getRulesDirectories(config.rulesDirectory, dir),
+            rulesDirectory,
         };
     }
 
-    function parseRules(config: RawRulesConfig | undefined): Map<string, Partial<IOptions>> {
+    function parseRules(config: RawRulesConfig | undefined,
+                        copyRulestoJsRules = false,
+                        rulesDirectory?: string | string[]): Map<string, Partial<IOptions>> {
```

Comment:
please move `copyRulesToJsRules` feature to a separate function, rather than co-opting this existing method into doing multiple things.